panic will happen if fencing a cluster which as no annotation defined

while fencing a cnp pod, if the cluster has no annotation defined, panic will report. 

it is lower chance people will remove annotation from cluster. this happens while create cluster using client-go client and the cluster created has no annotation in it. 

```
kcnpg fencing on postgresql-storage-class postgresql-storage-class-1 -n test2
panic: assignment to entry in nil map

goroutine 1 [running]:
github.com/cloudnative-pg/cloudnative-pg/pkg/utils.SetFencedInstances(0x55c7700?, 0xc000810018)
	/Users/tao/work/git/cloudnative-pg/cloudnative-pg/pkg/utils/fencing.go:86 +0x23c
github.com/cloudnative-pg/cloudnative-pg/pkg/utils.AddFencedInstance({0x7ff7bfeff0df, 0x1a}, 0xc0000460c8?)
	/Users/tao/work/git/cloudnative-pg/cloudnative-pg/pkg/utils/fencing.go:112 +0x1c7
github.com/cloudnative-pg/cloudnative-pg/internal/cmd/plugin/fence.ApplyFenceFunc({0x5af46b0, 0xc0000460c8}, {0x5afb360, 0xc0002e2c40}, {0x7ff7bfeff0c6, 0x18}, {0x7ff7bfeff0fd, 0x5}, {0x7ff7bfeff0df, 0x1a}, ...)
	/Users/tao/work/git/cloudnative-pg/cloudnative-pg/internal/cmd/plugin/fence/fence.go:80 +0x256
github.com/cloudnative-pg/cloudnative-pg/internal/cmd/plugin/fence.fencingOn({0x5af46b0?, 0xc0000460c8?}, {0x7ff7bfeff0c6?, 0x0?}, {0x7ff7bfeff0df, 0x1a})
	/Users/tao/work/git/cloudnative-pg/cloudnative-pg/internal/cmd/plugin/fence/fence.go:34 +0x85
github.com/cloudnative-pg/cloudnative-pg/internal/cmd/plugin/fence.glob..func1(0x68d5720, {0xc000457480?, 0x4?, 0x4?})
	/Users/tao/work/git/cloudnative-pg/cloudnative-pg/internal/cmd/plugin/fence/cmd.go:38 +0x116
github.com/spf13/cobra.(*Command).execute(0x68d5720, {0xc000457440, 0x4, 0x4})
	/Users/tao/work/git/cloudnative-pg/cloudnative-pg/vendor/github.com/spf13/cobra/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc000342f00)
	/Users/tao/work/git/cloudnative-pg/cloudnative-pg/vendor/github.com/spf13/cobra/command.go:990 +0x3b4
github.com/spf13/cobra.(*Command).Execute(...)
```

--- a/internal/cmd/plugin/fence/fence.go
+++ b/internal/cmd/plugin/fence/fence.go
@@ -76,7 +76,9 @@ func ApplyFenceFunc(
 	}
 
 	fencedCluster := cluster.DeepCopy()
-
+	if fencedCluster.Annotations == nil {
+		fencedCluster.Annotations = make(map[string]string)
+	}
 	if err = fenceFunc(serverName, fencedCluster.Annotations); err != nil {
 		return err
 	}

